### ISSUE TITLE
Default to costume tab for users wearing a costume

### ISSUE THREAD

**Issue by probaton** [HabitRPG/habitica#10434]
[//]: # (Before logging this issue, please post to the Report a Bug guild from the Habitica website's Help menu. Most bugs can be handled quickly there. If a GitHub issue is needed, you will be advised of that by a moderator or staff member -- a player with a dark blue or purple name. It is recommended that you don't create a new issue unless advised to.)

[//]: # (Bugs in the mobile apps can also be reported there.)

[//]: # (If you have a feature request, use "Help > Request a Feature", not GitHub or the Report a Bug guild.)

[//]: # (For more guidelines see https://github.com/HabitRPG/habitica/issues/2760)

[//]: # (Fill out relevant information - UUID is found from the Habitia website at User Icon > Settings > API)
### General Info
  * UUID: 6d7d6651-604d-4e7c-adff-a040770a6768
  * Browser: Chrome
  * OS: Fedora

### Description
[//]: # (Describe bug in detail here. Include screenshots if helpful.)
Every time I open the equipment page, the drawer at the bottom of the page defaults to the 'equipment' tab. This makes sense for starting users, but as a veteran player with a full set of quest gear, I honestly can't remember when I last (intentionally) made changes to my equipment. 

Of course the actual act of clicking the costume tab isn't necessarily a big problem. What vexes me is the fact that the indicator is tiny and I keep forgetting that my selection resets every time I open the equipment page. The result is that I regularly swap out two or three equipment items before realizing my mistake. I then have to spend a few minutes browsing through my huge wardrobe to find and re-equip my quest items. The fact that the equipment preview dialog displays the replacement item, even though my actual avatar is going to ignore it, is also very misleading. 

All this would go away if I could somehow convince Habitica to open my costume tab in the equipment drawer by default. I would suggest we simply open the costume tab by default for everyone who has the 'Use Costume' toggle set to true. Another option would be to remember the last selection in user settings somewhere, but I have to say that feels like overkill to me. 

#### Console Errors
[//]: # (Include any JavaScript console errors here.)

**Comment by probaton** [comment]
I don't mind taking a stab at this myself, but I'd love to hear what you feel is the best approach before I pick this up.

**Comment by probaton** [comment]
I accidentally edited my equipment again not five minutes after submitting this ticket while cobbling together my new pirate costume. D'oh.

**Comment by Alys** [comment]
I make that same mistake all the time too. I've just given up keeping my Battle Gear correct now - it's too much trouble.

I'd like to see this change made but @probaton you should wait for a staff member to give approval for this one before starting. I've marked it suggestion-discussion for now.

**Comment by probaton** [comment]
Might I coax someone to make an executive decision on this? As I said; I don't mind building this myself, and it seems like a pretty easy fix, but I wouldn't want to invest the time without an official go-ahead.

**Comment by paglias** [comment]
That's fine for me, @Tressley any objection?

**Comment by Tressley** [comment]
I don't think this is necessary. This is more of a power user request and useful only for users in the "end game." The typical user journey is centered around the introduction and acquisition of new pieces of equipment. Class level, quest rewards, Enchanted Armoire all introducing items that direct the user to change their Battle Armor/Costume.

With respect to the indicator, I have made design updates, but we still need someone to pick up the implementation. Here's a link to the issue: https://github.com/HabitRPG/habitica/issues/9458

**Comment by beffymaroo** [comment]
@Tressley Forgive me if I'm misunderstanding what's being discussed, but I don't see why having the tab default to "costume" if the user is using the costume option would be disruptive or unnecessary if it's an easy change to make. Users who are equipping costume items probably change their costume much more often than their battle gear, and users who are new or focused on battle gear alone will likely not run into a problem.

I'd also note that a lot of the items that are revenue-generating are most often equipped as costume items, such as the monthly subscriber gear which does not have any stats attached to it at all.

We do also get a lot of emails and social media questions (probably in Report a Bug as well, though I don't spend as much time in that area) because of confusion and frustration with not seeing your cool new item appear on your avatar when equipped.

**Comment by Tressley** [comment]
For me, while I do have my Costume enabled, I don't change my costume very often. I do, however, change my Battle Gear (still labeled "Equipment" on web) frequently as I like to bounce around the different classes. Maybe I misunderstood? Alys mentioned her Battle Gear being the issue, not her costume?

Because of the existing pattern, clicking "Equipment" and going to Costume isn't what I, as a user, expect. With that being said, in reading Probaton's initial comment I honed in on the indicator feedback and wanted to link the design changes that address this and the change in labeling (Equipment > Battle Gear.) Sure, it isn't as easy of a change, but I felt like it might have addressed what I read as the underlying problem of not knowing which one you're actually editing.

**Comment by beffymaroo** [comment]
@Tressley I for sure think updating the indicator is going to be really helpful!

Sounds like we have very different patterns of usage ;P I always have the Legendary gear set equipped for Battle Gear, and basically never change that, while I change my costume often. I think that seems to be true of a lot of folks whose social media posts I see and who post in the chats often- though I realize that is a relatively small subset of users (though it may include a large number of our subscribers).

I guess this is a situation where some analytics data would be helpful- I'm happy to take at least a quick dive into this this week if wanted.

**Comment by Alys** [comment]
> Alys mentioned her Battle Gear being the issue, not her costume?

My Battle Gear is an issue because I'm always changing it by accident when I want to change my Costume. I very rarely want to change Battle Gear.

While the improved indicator might help, it would still be most convenient if a player who had "use costume" turned on was automatically defaulted to the costume section. For those of us who experience this issue, the problem is not that we're unaware that we have costumes enabled, it's that we rarely think about changing Battle Gear and so when we're in the equipment screen we're automatically thinking about only our costume and don't remember the necessity to change to the costume section.

> useful only for users in the "end game." 

We do want those users to keep using Habitica though. I'm not sure we can really consider Habitica to have an end since ideally users would keep using it to manage tasks, and new equipment is released twice each month (plus seasonal gear).

**Comment by probaton** [comment]
Fixing the indicator would definitely be of help, but I have to say that even if it were better I would still appreciate the option of selecting the costume tab as default. 

Ever since I got a full set of epic gear I really haven't seen the point of changing battle gear, and I have to admit that I've assumed everyone else felt the same way. On the other hand, I change my costume pretty much every time I get new items from the armoire, my subscription, or the time travelers. No matter how good the indicator; that means I'm _going_ to forget to switch at some point, which is an exercise in frustration every time it happens. 

You have piqued my curiosity though, I'd love to hear what kind of power-player fine-tuning you do that has you updating your battle gear so often :) 

As for the equipment overhaul; it looks awesome, but I'm sorry to say it's far too big a task for something that far outside my comfort zone.

**Comment by Tressley** [comment]
Okay, so I spoke with some of the team and I think the best approach (for now) would be to have the page default to your last active tab. If you were editing your Costume, going back to Equipment will default to Costume. Same goes for Battle Gear.

How do we feel about that?

**Comment by probaton** [comment]
Sounds great. As I mentioned in my original request, I'm ok with both options. The main reason I was pushing the 'use costume' trigger was because it's easier to build. 

I'll take a stab at this as soon as I finish up processing @paglias ' feedback on my other open PR.
